File: dx-api/app/constants/game_status.go
package constants

// Game status values.
const (
	GameStatusDraft     = "draft"
	GameStatusPublished = "published"
	GameStatusWithdraw  = "withdraw"
)

// GameStatusLabels maps each status to its Chinese label.
var GameStatusLabels = map[string]string{
	GameStatusDraft:     "草稿",
	GameStatusPublished: "已发布",
	GameStatusWithdraw:  "已撤回",
}

// GameStatusOption represents a selectable game status.
type GameStatusOption struct {
	Value string
	Label string
}

// GameStatusOptions returns all game statuses as an ordered slice.
func GameStatusOptions() []GameStatusOption {
	return []GameStatusOption{
		{Value: GameStatusDraft, Label: "草稿"},
		{Value: GameStatusPublished, Label: "已发布"},
		{Value: GameStatusWithdraw, Label: "已撤回"},
	}
}

File: dx-web/src/consts/game-status.ts
export const GAME_STATUSES = {
  DRAFT: "draft",
  PUBLISHED: "published",
  WITHDRAW: "withdraw",
} as const;

export type GameStatus = (typeof GAME_STATUSES)[keyof typeof GAME_STATUSES];

export const GAME_STATUS_LABELS: Record<GameStatus, string> = {
  draft: "草稿",
  published: "已发布",
  withdraw: "已撤回",
};

export const GAME_STATUS_OPTIONS: { value: GameStatus; label: string }[] =
  Object.entries(GAME_STATUS_LABELS).map(([value, label]) => ({
    value: value as GameStatus,
    label,
  }));
